### PR TITLE
Fix invalid phpdoc

### PR DESCRIPTION
**Pull request by mvorisek**
fixes most phpstan level 2 issues

and upgrade phpstan to level 6 to prevent code degradation from new contributions

no API change

### CODE REVIEW

**Comment by mvorisek** [comment]
@AllanJard can these fixes be merged?

there should be no functional change except fixed imports in several catch clauses like in https://github.com/DataTables/Editor-PHP/pull/17/files#diff-01b189458d10702351d63e3a6e1e0d5e9344feae55742cceab46b48cc5ddba6fR102

**Comment by mvorisek** [comment]
@AllanJard is now a better time? We get a lot of phpstan issues because of phpdoc issues fixed with this PR. Thanks in advance.

**Comment by AllanJard** [comment]
Apologies - I've got a massive backlog of stuff at the moment and haven't been able to prioritise this like I want to. It is very much on my radar though and everything looks fine on a quick scan.

**Comment by AllanJard** [comment]
This is in part a note to myself for when I get to looking at this - [this thread in the forum](https://datatables.net/forums/discussion/72033/static-function-query-connect-should-not-be-abstract#latest) notes an issue with the abstract clarification from 6ce4b0a9fe8699ec178e89a17fbf7442a523eb84. It is the right thing to do, but it introduced an error with 5.6.0.

Although 5.6 is EOL and hasn't been supported by PHP since 2018, many are stuck with it. I'm concerned about this, so as a note to me - check changes with back to 5.4, which is what we say we currently support.

**Comment by mvorisek** [comment]
@AllanJard is it now a good time to fix the phpdoc documentation issues?

The correct phpdocs are important for static analysis for dependent projects.

They are several conflicts since the last push, please let me know if we can coordinate the merge soon, and I will fix them. Thanks.

**Comment by AllanJard** [comment]
Apologies - I'm just a little bit swamped! I need to go over them all with a linter when I get a chance. Not yet sure when that will be.

**Comment by mvorisek** [comment]
@AllanJard What to do with this PR? When I will fix the conflicts (rebase on the latest master), can I expect it to be merged then?

File: Editor.php [hunk 1]
```diff
@@ -600,7 +599,7 @@ public function pkey ( $_=null )
 	 *
```
Comment (mvorisek) [inline Editor.php:599]: TODO for myself - `pkey` method returns `array`

File: Editor.php [hunk 2]
```diff
@@ -600,7 +599,7 @@ public function pkey ( $_=null )
...
 	 * @param  string  $row   The row of data that the primary key value should
```
Comment (mvorisek) [inline Editor.php:600]: TODO for myself - should accept array

File: Editor.php [hunk 3]
```diff
@@ -341,9 +340,9 @@ public function field ( $_=null )
 
 	/**
 	 * Get / set field instances.
-	 * 
+	 *
 	 * An alias of {@see field}, for convenience.
-	 *  @param Field $_... Instances of the {@see Field} class, given as a single 
+	 *  @param Field $_... Instances of the {@see Field} class, given as a single
```
Comment (mvorisek) [inline Editor.php:345]: TODO for myself, should this accept array as the 1st param?

File: Database.php [hunk 4]
```diff
@@ -13,19 +13,18 @@
...
  *
  * Create a database connection which may then have queries performed upon it.
- * 
+ *
  * This is a database abstraction class that can be used on multiple different
  * databases. As a result of this, it might not be suitable to perform complex
- * queries through this interface or vendor specific queries, but everything 
+ * queries through this interface or vendor specific queries, but everything
  * required for basic database interaction is provided through the abstracted
  * methods.
```
Comment (mvorisek) [inline Database.php:29]: TODO for myself - `_constructor` must accept `array{ pdo: \PDO }`

**Comment by AllanJard** [comment]
I'm really sorry this has slipped by me. There is just so much going on. I'm working on the 2.2 cycle at the moment, so if you are willing to rebase now, that would be great.

**Comment by mvorisek** [comment]
@AllanJard once done, may I ask for one DT Editor license for free in exchange for this work?

**Comment by AllanJard** [comment]
Yes, that seems like a fair trade :-)

**Comment by mvorisek** [comment]
@AllanJard PR is done

**Comment by AllanJard** [comment]
Amazing - thank you.

**Pull request merged**